File: Minimum Depth of Binary Tree/MinimumDepthofBinaryTree.cs
using System;

namespace LeetcodePracticeCsharpVersion
{
    class MinimumDepthofBinaryTree
    {
        public int MinDepth(TreeNode root)
        {
            if (root == null) return 0;
            int left = MinDepth(root.left);
            int right = MinDepth(root.right);

            return (left == 0 || right == 0) ? left + right + 1 : Math.Min(left, right) + 1;
        }
    }
}

File: Minimum Depth of Binary Tree/MinimumDepthofBinaryTree.py
class MinimumDepthofBinaryTree(object):
    def minDepth(self, root):
        if root is None:
            return 0
        if not root.left or not root.right:
            return self.minDepth(root.left) + self.minDepth(root.right) + 1
        return min(self.minDepth(root.left), self.minDepth(root.right)) + 1